[2024-11-23T17:08:46.143+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-11-23T17:08:46.178+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: fraud_detection.process_transaction scheduled__2024-11-23T17:07:00+00:00 [queued]>
[2024-11-23T17:08:46.193+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: fraud_detection.process_transaction scheduled__2024-11-23T17:07:00+00:00 [queued]>
[2024-11-23T17:08:46.194+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 2
[2024-11-23T17:08:46.218+0000] {taskinstance.py:2888} INFO - Executing <Task(BranchPythonOperator): process_transaction> on 2024-11-23 17:07:00+00:00
[2024-11-23T17:08:46.235+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=1362) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-11-23T17:08:46.237+0000] {standard_task_runner.py:72} INFO - Started process 1367 to run task
[2024-11-23T17:08:46.239+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'fraud_detection', 'process_transaction', 'scheduled__2024-11-23T17:07:00+00:00', '--job-id', '138', '--raw', '--subdir', 'DAGS_FOLDER/fraud_detection_dag.py', '--cfg-path', '/tmp/tmpf1w2kkrr']
[2024-11-23T17:08:46.244+0000] {standard_task_runner.py:105} INFO - Job 138: Subtask process_transaction
[2024-11-23T17:08:46.331+0000] {task_command.py:467} INFO - Running <TaskInstance: fraud_detection.process_transaction scheduled__2024-11-23T17:07:00+00:00 [running]> on host 345e92ff26bf
[2024-11-23T17:08:46.560+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='fraud_team' AIRFLOW_CTX_DAG_ID='fraud_detection' AIRFLOW_CTX_TASK_ID='process_transaction' AIRFLOW_CTX_EXECUTION_DATE='2024-11-23T17:07:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-11-23T17:07:00+00:00'
[2024-11-23T17:08:46.563+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-11-23T17:08:46.586+0000] {baseoperator.py:405} WARNING - BranchPythonOperator.execute cannot be called outside TaskInstance!
[2024-11-23T17:08:46.612+0000] {fraud_detection_dag.py:78} INFO - Données récupérées depuis XCom: {"columns":["cc_num","merchant","category","amt","first","last","gender","street","city","state","zip","lat","long","city_pop","job","dob","trans_num","merch_lat","merch_long","is_fraud","current_time"],"index":[276962],"data":[[4956828990005111019,"fraud_Mante, Luettgen and Hackett","health_fitness",44.72,"Kenneth","Robinson","M","269 Sanchez Rapids","Elizabeth","NJ",7208,40.6747,-74.2239,124967,"Operational researcher","1980-12-21","fb126f8dbf76c4bf24d75695b371dcf4",39.725766,-74.965442,0,1732381722714]]}
[2024-11-23T17:08:46.612+0000] {fraud_detection_dag.py:86} INFO - Données JSON parsées avec succès
[2024-11-23T17:08:46.618+0000] {fraud_detection_dag.py:90} INFO - DataFrame créé avec 1 lignes
[2024-11-23T17:08:46.621+0000] {fraud_detection_dag.py:94} INFO - Colonnes renommées avec succès
[2024-11-23T17:08:46.622+0000] {fraud_detection_dag.py:95} INFO - Colonnes disponibles: ['cc_num', 'merchant', 'category', 'amt', 'first', 'last', 'gender', 'street', 'city', 'state', 'zip', 'lat', 'long', 'city_pop', 'job', 'dob', 'trans_num', 'merch_lat', 'merch_long', 'is_fraud', 'trans_date_trans_time']
[2024-11-23T17:08:46.627+0000] {fraud_detection_dag.py:99} INFO - Conversion de trans_date_trans_time en unix_time réussie
[2024-11-23T17:08:46.668+0000] {fraud_detection_dag.py:108} INFO - Transformation des données...
[2024-11-23T17:08:47.987+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/sklearn/ensemble/_base.py:172: UserWarning: Loky-backed parallel loops cannot be called in a multiprocessing, setting n_jobs=1
  n_jobs = min(effective_n_jobs(n_jobs), n_estimators)

[2024-11-23T17:08:48.009+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/sklearn/ensemble/_base.py:172: UserWarning: Loky-backed parallel loops cannot be called in a multiprocessing, setting n_jobs=1
  n_jobs = min(effective_n_jobs(n_jobs), n_estimators)

[2024-11-23T17:08:48.018+0000] {fraud_detection_dag.py:120} INFO - Prediction de fraude: NORMAL, Probabilité: 31.21%
[2024-11-23T17:08:48.097+0000] {fraud_detection_dag.py:126} INFO - Prédiction: NORMAL (Probabilité: 31.21%)
[2024-11-23T17:08:48.101+0000] {python.py:240} INFO - Done. Returned value was: notify_normal
[2024-11-23T17:08:48.102+0000] {branch.py:38} INFO - Branch into notify_normal
[2024-11-23T17:08:48.103+0000] {skipmixin.py:230} INFO - Following branch ('notify_normal',)
[2024-11-23T17:08:48.124+0000] {skipmixin.py:278} INFO - Skipping tasks [('notify_fraud', -1), ('skip_processing', -1)]
[2024-11-23T17:08:48.193+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-11-23T17:08:48.194+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=fraud_detection, task_id=process_transaction, run_id=scheduled__2024-11-23T17:07:00+00:00, execution_date=20241123T170700, start_date=20241123T170846, end_date=20241123T170848
[2024-11-23T17:08:48.279+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2024-11-23T17:08:48.350+0000] {taskinstance.py:3900} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2024-11-23T17:08:48.368+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
